File: 8-firstNonRecurringChar.py
# Question variation: https://www.youtube.com/watch?v=GJdiM-muYqc
#
# Give the first NON repeating letter given a string of random letters
#
# Example: DBCABA -> B

string = "AABBECC"
counter = {}

def first_non_recurring_char_in(given_string):
  for char in given_string:
    if char in counter:
      counter[char] += 1
    else:
      counter[char] = 1
  
  for char in given_string:
    if counter[char] == 1:
      return char
  
  return None

result = first_non_recurring_char_in(string)
print(result)

File: 8-firstNonRecurringChar.swift
// Question variation: https://www.youtube.com/watch?v=GJdiM-muYqc
//
// Give the first NON repeating letter given a string of random letters
//
// Example: DBCABA -> B

import Foundation

var string = "DBCABA"
var counter: [String: Int] = [:]


func returnFirstNonRepeatingChar(in givenString: String) -> String? {
    for character in givenString {
        let key = String(character)
        if counter[key] != nil {
            counter[key]! += 1
        } else {
            counter[key] = 1
        }
    }

    for character in givenString {
        let key = String(character)
        if let count = counter[key], count == 1 {
            return key
        }
    }
    return nil
}

var result = returnFirstNonRepeatingChar(in: string)
print("First non repeating char is \(result ?? "0")")